bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,135 @@
+        """
+        Convert a relative or absolute date/time string to an arrow.Arrow object.
+
+        Try to interpret the date string as a relative time. If conversion fails, try to interpret it as an absolute
+        time. Tokens that are not recognised are returned along with the part of the string that was successfully
+        converted to an arrow object. If the date string cannot be parsed, BadArgument is raised.
+        """
+        try:
+            return arrow.utcnow().dehumanize(argument)
+        except ValueError:
+            try:
+                dt, ignored_tokens = parser.parse(argument, fuzzy_with_tokens=True)
+            except parser.ParserError:
+                raise commands.BadArgument(f"`{argument}` Could not be parsed to a relative or absolute date.")
+            except OverflowError:
+                raise commands.BadArgument(f"`{argument}` Results in a date outside of the supported range.")
+            return arrow.get(dt), ignored_tokens
+
+
+class Epoch(commands.Cog):
+    """Convert an entered time and date to a unix timestamp."""
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: DateString = None) -> None:
+        """
+        Convert an entered date/time string to the equivalent epoch.
+
+        **Relative time**
+            Must begin with `in...` or end with `...ago`.
+            Accepted units: "seconds", "minutes", "hours", "days", "weeks", "months", "years".
+            eg `.epoch in a month 4 days and 2 hours`
+
+        **Absolute time**
+            eg `.epoch 2022/6/15 16:43 -04:00`
+            Absolute times must be entered in descending orders of magnitude.
+            If AM or PM is left unspecified, the 24-hour clock is assumed.
+            Timezones are optional, and will default to UTC. The following timezone formats are accepted:
+                Z (UTC)
+                ±HH:MM
+                ±HHMM
+                ±HH
+
+        Times in the dropdown are shown in UTC
+        """
+        if not date_time:
+            await invoke_help_command(ctx)
+            return
+
+        if isinstance(date_time, tuple):
+            # Remove empty strings. Strip extra whitespace from the remaining items
+            ignored_tokens = list(map(str.strip, filter(str.strip, date_time[1])))
+            date_time = date_time[0]
+            if ignored_tokens:
+                await ctx.send(f"Could not parse the following token(s): `{', '.join(ignored_tokens)}`")
+            await ctx.send(f"Date and time parsed as: `{date_time.format(arrow.FORMAT_RSS)}`")
+
+        epoch = int(date_time.timestamp())
+        view = TimestampMenuView(ctx, self._format_dates(date_time), epoch)
+        original = await ctx.send(f"`{epoch}`", view=view)
+        if await view.wait():  # wait until expiration before removing the dropdown
+            await original.edit(view=None)
```

Comment:
this can error if the message has somehow been deleted when we try to delete it. 

Might not worth be changing, given that there's not a wait_for_deletion or anything, so only a mod can delete it.